pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -378,24 +378,14 @@ func translateRoute(push *model.PushContext, node *model.Proxy, in *networking.H
 			}
 		}
 
-		if len(in.AppendHeaders) > 0 {
-			action.RequestHeadersToAdd = make([]*core.HeaderValueOption, 0)
-			for key, value := range in.AppendHeaders {
-				action.RequestHeadersToAdd = append(action.RequestHeadersToAdd, &core.HeaderValueOption{
-					Header: &core.HeaderValue{
-						Key:   key,
-						Value: value,
-					},
-				})
-			}
-		}
+		// TODO: deprecate this
+		action.RequestHeadersToAdd = translateAppendHeaders(in.AppendHeaders)
```

Comment:
just remove it.. There is no need to deprecate.. Its alredy deprecated in envoy. Instead, add this to every entry in the weighted cluster